src/components/LocationsSearch/LocationsSearch.jsx

```diff
@@ -0,0 +1,52 @@
+import PropTypes from 'prop-types';
+import { useState } from 'react';
+import { CurrentLocationInfoType } from '../../types/types';
+import Preloader from '../Preloader/Preloader';
+import './LocationsSearch.css';
+import LocationsSearchReuslts from './LocationsSearchReuslts/LocationsSearchReuslts';
+import debounce from 'lodash/debounce';
+import { DEBOUNCE_WAIT_TIME, MIN_CHARACKTERS_FOR_SEARCH } from '../../constants/constants';
+
+function LocationsSearch({ isReadyForSearch, searchResults, clearSearch, searchLocations }) {
+  const [inputValue, setInputValue] = useState('');
+  const debouncedSearchLocations = debounce(searchLocations, DEBOUNCE_WAIT_TIME);
+
+  function handleInputChange(event) {
+    const value = event.target.value;
+    setInputValue(value);
+
+    if (value === '') {
+      clearSearch();
```

Comment:
can you elaborate why do you need to clear something out if value itself is already empty?

Reply:
The logics is following:
We need to clear search results if user clears search input.
Let's imagine that user searched something previously - the results of this search are currently shown.
If user wants to start a new search he or she deletes all text in the search input (users often do this before new search) - results should disappear at this moment. In other case user will continue looking at the old results thinking that they are new...

Reply:
> The logics is following: We need to clear search results if user clears search input. Let's imagine that user searched something previously - the results of this search are currently shown. If user wants to start a new search he or she deletes all text in the search input (users often do this before new search) - results should disappear at this moment. In other case user will continue looking at the old results thinking that they are new...

You need to use this function because you send search request after value.length >= MIN_CHARACTERS_FOR_SEARCH. If you didn't use this condition, you wouldn't use this function.

Reply:
> You need to use this function because you send search request after value.length >= MIN_CHARACTERS_FOR_SEARCH. If you didn't use this condition, you wouldn't use this function.
It's senseless to remove this condition because it will cause excessive network load (Server answers only if the condition is true). Also it's weird to send request with empty query.

Reply:
We do not want to enlarge user waiting in any case so it's much faster to check the condition on client's side than send server request and wait for it (with slow 3G it can take several seconds). So I still think that I need this condition here.
Also debounce is already used in couple with the condition check.
Thanks a lot.